src/hotspot/share/gc/shenandoah/shenandoahFreeSet.hpp

```diff
@@ -28,63 +30,186 @@
 #include "gc/shenandoah/shenandoahHeapRegionSet.hpp"
 #include "gc/shenandoah/shenandoahHeap.hpp"
 
-class ShenandoahFreeSet : public CHeapObj<mtGC> {
+enum ShenandoahFreeMemoryType : uint8_t {
+  NotFree,
```

Comment:
Thanks for your feedback.  I'm adding comments on these declarations.

Replacing ShenandoahFreeMemoryType with ShenandoahFreeSetRegionType.

Reply:
Sorry I wasn't clear. My question was about the "Free" in the name of the type, as if it represented subtypes of "FreeRegions". But since it includes "NotFree", it includes regions that aren't free. May be "RegionFreenessType" if you want free in the name. Then "NotFree" is lack of freeness :-)

But may be I am nitpicking too much here.